### ISSUE TITLE
Bug in Odoo gamification addon

### ISSUE THREAD

**Issue by salahzidi** [odoo/odoo#8545]
When I create a goal definition with: computation mode : (Automatic: sum on a field, Model : Sale Order Line, Field to Sum : SubTotal)

When I add this goal definition to a challenge and try to start it, I get this traceback:
```
Client Traceback (most recent call last):
  File "/home/openerp/server/openerp/addons/web/http.py", line 204, in dispatch
    response["result"] = method(self, **self.params)
  File "/home/openerp/server/openerp/addons/web/controllers/main.py", line 1132, in call_button
    action = self._call_kw(req, model, method, args, {})
  File "/home/openerp/server/openerp/addons/web/controllers/main.py", line 1120, in _call_kw
    return getattr(req.session.model(model), method)(*args, **kwargs)
  File "/home/openerp/server/openerp/addons/web/session.py", line 42, in proxy
    result = self.proxy.execute_kw(self.session._db, self.session._uid, self.session._password, self.model, method, args, kw)
  File "/home/openerp/server/openerp/addons/web/session.py", line 30, in proxy_method
    result = self.session.send(self.service_name, method, *args)
  File "/home/openerp/server/openerp/addons/web/session.py", line 103, in send
    raise xmlrpclib.Fault(openerp.tools.ustr(e), formatted_info)

Server Traceback (most recent call last):
  File "/home/openerp/server/openerp/addons/web/session.py", line 89, in send
    return openerp.netsvc.dispatch_rpc(service_name, method, args)
  File "/home/openerp/server/openerp/netsvc.py", line 296, in dispatch_rpc
    result = ExportService.getService(service_name).dispatch(method, params)
  File "/home/openerp/server/openerp/service/web_services.py", line 626, in dispatch
    res = fn(db, uid, *params)
  File "/home/openerp/server/openerp/osv/osv.py", line 190, in execute_kw
    return self.execute(db, uid, obj, method, *args, **kw or {})
  File "/home/openerp/server/openerp/osv/osv.py", line 132, in wrapper
    return f(self, dbname, *args, **kwargs)
  File "/home/openerp/server/openerp/osv/osv.py", line 199, in execute
    res = self.execute_cr(cr, uid, obj, method, *args, **kw)
  File "/home/openerp/server/openerp/osv/osv.py", line 187, in execute_cr
    return getattr(object, method)(cr, uid, *args, **kw)
  File "/home/openerp/server/openerp/addons/gamification/plan.py", line 342, in action_start
    return self.generate_goals_from_plan(cr, uid, ids, context=context)
  File "/home/openerp/server/openerp/addons/gamification/plan.py", line 439, in generate_goals_from_plan
    goal_obj.update(cr, uid, [new_goal_id], context=context)
  File "/home/openerp/server/openerp/addons/gamification/goal.py", line 278, in update
    new_value = res and res[0][field_name] or 0.0
KeyError: u'price_subtotal'
```
this is a video that shows the scenario that I did: http://youtu.be/b546xTc1ArY

**Comment by Yenthe666** [comment]
Hi @salahzidi,

Thank you for reporting this issue. This issue still seems to be present in Odoo 8.
My goal definition:
![image](https://cloud.githubusercontent.com/assets/6352350/15531842/d912d94c-225c-11e6-8a3b-15ebcce3352d.png)


My challenge configuration:
![image](https://cloud.githubusercontent.com/assets/6352350/15531828/c39eec90-225c-11e6-9598-bbb0e3bf31e2.png)
![image](https://cloud.githubusercontent.com/assets/6352350/15531832/ca35bbce-225c-11e6-83e8-1378182b902b.png)
@JKE-be this might be one for you? Since it is partially website, not sure though.

Regards,
Yenthe

**Comment by JKE-be** [comment]
Hello @Yenthe666,

I just take a look on it.
The problem is because you should not be able to create Goals with some functional/computed fields which are not stored (Read_froup cannot work with un-stored field).
This module was developed by @mart-e, so I will let him choose the fix, but I think that the best patch in stable version is to add domain on field and only display stored field 'field_id' ~= Field to sum!

**Comment by Yenthe666** [comment]
Alright thanks @JKE-be! I agree that a domain would probably be the cleanest and easiest fix for this.

**Comment by mart-e** [comment]
Fixed at 97492a12a9f0975b80ebd43422aeebfb315dacb2
Thanks

**Comment by CarlesAntoli** [comment]
With the last commit 8e7ad61 of archive goal.py i get the following error:

"File "/opt/odoo/server/addons/gamification/models/goal.py", line 160, in _check_model_validity
    raise UserError(_("The model configuration for the definition %s seems incorrect, please check it.\n\n%s not found") % (definition.name, e.message))
ParseError: "global name 'UserError' is not defined" while parsing /opt/odoo/Extra_Addons/utilidades/gamification_crm_goals/opportunity_crm_goals.xml:45, near
<record model="gamification.goal.definition" id="definition_crm_historico_tasa_exito">
            <field name="name">Porcentaje de &#233;xito (hist&#243;rico)</field>
            <field name="description">Hist&#243;rico del porcentaje de &#233;xito de oportunidades ganadas sobre perdidas</field>
            <field name="computation_mode">python</field>
            <field name="compute_code">def get_porcentaje_historico(cr, uid, context=None):
                total_op_gan= pool.get('crm.lead').search(cr, uid, [('user_id','=',object.user_id.id),('type', '=', 'opportunity'),     ('stage_id','=',6)], count=True, context=context) * 100

                total_op_gan_per= pool.get('crm.lead').search(cr, uid, [('user_id','=',object.user_id.id),('type', '=', 'opportunity'),('stage_id','in',(6, 7))], count=True, context=context)
                val_porcentaje_historico=0
                try:
                 val_porcentaje_historico= total_op_gan/ total_op_gan_per
                except:
	         val_porcentaje_historico==0
                 return val_porcentaje_historico
                else:
                 return val_porcentaje_historico

result = get_porcentaje_historico(cr, uid, context=context)</field>
            <field name="condition">higher</field>
            <field name="suffix">%</field>
            <field name="model_id" eval="ref('crm.model_crm_lead')"/>
        </record>"

It is a bug or a problem of my code, thanks

**Comment by mart-e** [comment]
Hello,

Yes there were an error in my 8.0 commit a4687c6e42a7b0ec6b8d98dc2af7fe2d1c3272d0, it is fixed at a4687c6e42a7b0ec6b8d98dc2af7fe2d1c3272d0 and 37bbe87d5792c879130e5f1e491f743cbc50dc1e
Thanks for the report
